Mathlib/Topology/Separation.lean

```diff
@@ -1311,6 +1311,10 @@ theorem Bornology.relativelyCompact_eq_inCompact [T2Space α] :
   Bornology.ext _ _ Filter.coclosedCompact_eq_cocompact
 #align bornology.relatively_compact_eq_in_compact Bornology.relativelyCompact_eq_inCompact
 
+theorem IsCompact.preimage_continuous [CompactSpace α] [T2Space β] {f : α → β} {s : Set β}
+    (hs : IsCompact s) (hf : Continuous f) : IsCompact (f ⁻¹' s) :=
+  (hs.isClosed.preimage hf).isCompact
```

Comment:
While you're at it, could you link this to the [IsProperMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ProperMap.html#IsProperMap) API by showing that any map from a compact space to a Hausdorff space is proper?

Reply:
BTW, we have `IsProperMap` and `CocompactMap`.

Reply:
I know, I added `IsProperMap` recently. I claim that it is the right concept mathematically, so I think we should expand this API rather than that of `CocompactMap`.

Reply:
> ... right concept mathematically

I wouldn't be surprised if both are useful (or even all 3 if you add `isCompact_preimage`). BTW, I didn't look closely at the definitions; which one is weaker?

Reply:
In general spaces (and for continuous functions of course), `IsProperMap` -> `isCompact_preimage` -> `CocompactMap` . The equivalence between `isCompact_preimage` and `CocompactMap` works as soon as the codomain is Hausdorff, and everything is equivalent when the codomain is Hausdorff and locally compact.

Regarding `isCompact_preimage` vs `CocompactMap`, I think we can safely redefine `CocompactMap` in terms of compact preimages, the tendsto definition was a nice trick but I don't think it's mathematically meaningful.

`IsProperMap` on the other hand is indeed significantly stronger than `CocompactMap`, so I'm not suggesting removing the latter, but I know that some key results that are usually proved for `isCompact_preimage` maps only work in locally compact space unless you assume the stronger `IsProperMap`. I don't have a good example in mathlib of such a theorem about `CocompactMap`, but we have some in the context of group actions. For example you can remove the locally compactness assumption in [t2Space_of_properlyDiscontinuousSMul_of_t2Space](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/ConstMulAction.html#t2Space_of_properlyDiscontinuousSMul_of_t2Space) if you strengthen the definition of [ProperlyDiscontinuousSMul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/ConstMulAction.html#ProperlyDiscontinuousSMul) to use `IsProperMap`.

Reply:
Done. I also added a version of `isProperMap_iff_ultrafilter` for a T₂ codomain.

Reply:
Do you think I should use `IsProperMap` instead of `IsCompact s → IsCompact (f ⁻¹' s)`.